armi/reactor/converters/uniformMesh.py

```diff
@@ -105,27 +106,74 @@ def __init__(self, cs=None):
         if r is None:
             raise ValueError(f"No reactor provided in {self}")
 
-        runLog.extra("Building copy of {} with a uniform axial mesh".format(r))
+        runLog.extra(f"Building with a uniform axial mesh reactor from {r}")
         completeStartTime = timer()
         self._sourceReactor = r
-        self.convReactor = self.initNewReactor(r)
-        self._setParamsToUpdate()
-        self._computeAverageAxialMesh()
-        self._buildAllUniformAssemblies()
-        self._mapStateFromReactorToOther(
-            self._sourceReactor, self.convReactor, mapNumberDensities=False
-        )
-        self.convReactor.core.updateAxialMesh()
 
-        self._newAssembliesAdded = self.convReactor.core.getAssemblies()
+        # Here we are taking a short cut to homogenizing the core by only focusing on the
+        # core assemblies that need to be homogenized. This will have a large speed up
+        # since we don't have to create an entirely new reactor perform the data mapping.
+        # We are using the Spent Fuel Pool (SFP) as a container just to temporarily store
+        # the detailed assembly so that it can be recovered later when the geometry
+        # conversions are undone.
+        if self._hasNonUniformAssems:
+            self.convReactor = self._sourceReactor
+            self._setParamsToUpdate()
+            for assem in self.convReactor.core.getAssemblies(self._nonUniformMeshFlags):
+                homogAssem = self.makeAssemWithUniformMesh(
+                    assem,
+                    self.convReactor.core.refAssem.getAxialMesh(),
+                    self.blockParamNames,
+                )
+                homogAssem.spatialLocator = assem.spatialLocator
+
+                # Remove this assembly from the core and add it to the
+                # SFP so that it can be replaced with the homogenized assembly.
+                # Note that we do not call the `removeAssembly` method because
+                # this will delete the core assembly from existence rather than
+                # only stripping its spatialLocator away.
+                if assem.spatialLocator in self.convReactor.core.childrenByLocator:
+                    self.convReactor.core.childrenByLocator.pop(assem.spatialLocator)
+                self.convReactor.core.remove(assem)
+                self.convReactor.core.assembliesByName.pop(assem.getName(), None)
+                for b in assem:
+                    self.convReactor.core.blocksByName.pop(b.getName(), None)
+
+                assem.setName(assem.getName() + self._SFP_TEMP_STORAGE_SUFFIX)
+                self.convReactor.core.sfp.add(assem)
+                self.convReactor.core.add(homogAssem)
+
+            self.convReactor.core.updateAxialMesh()
+
+            self._checkConversion()
+            completeEndTime = timer()
+            runLog.extra(
+                f"Reactor core conversion time: {completeEndTime-completeStartTime} seconds"
+            )
```

Comment:
We should just use the runLog after the if/else

Reply:
Yeah that's a good idea

Reply:
Taking it one step further - it looks like this block can be safely moved out of the if/else. 

```python
self.convReactor.core.updateAxialMesh()
self._checkConversion()
completeEndTime = timer()
runLog.extra(
    f"Reactor core conversion time: {completeEndTime-completeStartTime} seconds"
)
```